tools/testsys/src/aws_resources.rs

```diff
@@ -0,0 +1,231 @@
+                name: Some(cluster_name.clone()),
+                namespace: Some(NAMESPACE.into()),
+                labels: Some(self.labels()),
+                ..Default::default()
+            },
+            spec: ResourceSpec {
+                depends_on: None,
+                agent: Agent {
+                    name: "eks-provider".to_string(),
+                    image: testsys_images.eks_resource.clone(),
+                    pull_secret: testsys_images.secret.clone(),
+                    keep_running: false,
+                    timeout: None,
+                    configuration: Some(
+                        EksClusterConfig {
+                            cluster_name,
+                            creation_policy: Some(CreationPolicy::IfNotExists),
+                            region: Some(self.region.clone()),
+                            zones: None,
+                            version: Some(cluster_version),
+                            assume_role: self.assume_role.clone(),
+                        }
+                        .into_map()
+                        .context("Unable to convert eks config to map")?,
+                    ),
+                    secrets: self.secrets.clone(),
+                    capabilities: None,
+                },
+                destruction_policy: DestructionPolicy::Never,
+            },
+            status: None,
+        };
+        Ok(Crd::Resource(eks_crd))
+    }
+
+    fn ec2_crd(&self, cluster_suffix: &str, testsys_images: &TestsysImages) -> Result<Crd> {
+        let cluster_name = self.cluster_name(cluster_suffix);
+        let mut ec2_config = Ec2Config {
+            node_ami: self.ami.clone(),
+            // TODO - configurable
+            instance_count: Some(2),
+            instance_type: self.instance_type.clone(),
+            cluster_name: format!("${{{}.clusterName}}", cluster_name),
+            region: format!("${{{}.region}}", cluster_name),
+            instance_profile_arn: format!("${{{}.iamInstanceProfileArn}}", cluster_name),
+            subnet_id: format!("${{{}.privateSubnetId}}", cluster_name),
+            cluster_type: ClusterType::Eks,
+            endpoint: Some(format!("${{{}.endpoint}}", cluster_name)),
+            certificate: Some(format!("${{{}.certificate}}", cluster_name)),
+            cluster_dns_ip: Some(format!("${{{}.clusterDnsIp}}", cluster_name)),
+            security_groups: vec![],
+            assume_role: self.assume_role.clone(),
+        }
+        .into_map()
+        .context("Unable to create ec2 config")?;
+
+        // TODO - we have change the raw map to reference/template a non string field.
+        let previous_value = ec2_config.insert(
+            "securityGroups".to_owned(),
+            Value::String(format!("${{{}.securityGroups}}", cluster_name)),
+        );
+        if previous_value.is_none() {
+            todo!("This is an error: fields in the Ec2Config struct have changed")
+        }
```

Comment:
I'm confused by this block. IIUC, even though we replaced the previous value, we care if it was empty. If we are planning to handle this in a better way in the future I think the error in the [`todo!`](https://doc.rust-lang.org/src/core/macros/mod.rs.html#770-777) macro should be clearer since the resulting panic will print something like:

```
not yet implemented: This is an error: fields in the Ec2Config struct have changed
```

Or in case we won't handle this case, then an Error should be better instead of `panic`'ing the process.